chartfx-dataset/src/main/java/de/gsi/dataset/spi/DoubleDataSet3D.java

```diff
@@ -117,23 +117,34 @@ public String getStyle(final int index) {
     }
 
     @Override
-    public double getX(final int i) {
-        return xValues[i];
-    }
-
-    @Override
-    public double[] getXValues() {
-        return Arrays.copyOf(xValues, xValues.length);
+    public final double get(final int dimIndex, final int index) {
+        switch (dimIndex) {
+        case DIM_X:
+            return xValues[index];
+        case DIM_Y:
+            return yValues[index];
+        case DIM_Z:
+            return zValues[index / xValues.length][index % xValues.length];
+        }
+        return Double.NaN;
     }
 
     @Override
-    public double getY(final int i) {
-        return yValues[i];
+    public double getX(final int i) {
+        return xValues[i];
     }
 
     @Override
-    public double[] getYValues() {
-        return Arrays.copyOf(yValues, yValues.length);
+    public double[] getValues(final int dimIndex) {
+        switch (dimIndex) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/app/GSI/chart-fx/pullRequest?prid=4477772)